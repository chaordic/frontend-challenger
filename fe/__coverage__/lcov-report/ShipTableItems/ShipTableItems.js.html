<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for ShipTableItems/ShipTableItems.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">ShipTableItems</a> ShipTableItems.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Fragment } from 'react';
import { array, object } from 'prop-types';
import cssModules from 'react-css-modules';
import styles from './styles.scss';
&nbsp;
/**
 * Format to portuguese shop status
 *
 * @param {Array} items Products
 * @param {Object} geral Object with consolidate data by items
 * @return {Component} Table of products
 */
const ShipTableItems = ({ items, geral }) =&gt; (
  &lt;Fragment&gt;
    &lt;h1&gt;Detalhes da entrega&lt;/h1&gt;
    &lt;table className={styles.table}&gt;
      &lt;thead className={`${styles.background}`}&gt;
        &lt;tr className={styles.tr}&gt;
          &lt;td&gt;PRODUTO&lt;/td&gt;
          &lt;td&gt;SKU&lt;/td&gt;
          &lt;td&gt;QTD.&lt;/td&gt;
          &lt;td&gt;PREÃ‡O&lt;/td&gt;
        &lt;/tr&gt;
      &lt;/thead&gt;
      &lt;tbody&gt;
        {items.map((item, i) =&gt; (
          &lt;tr key={i} className={`${styles.tr} ${(i + 1) % 2 === 0 ? `${styles.background}` : ''}`}&gt;
            &lt;td className={styles.mainProduct}&gt;
              &lt;img
                alt={item.name}
                title={item.name}
                src={item.image}
              /&gt;
              &lt;section&gt;
                &lt;p className={`${styles.address}`}&gt;
                  {item.name}
                &lt;/p&gt;
&nbsp;
                &lt;p className={`${styles.address}`}&gt;
                  {item.desc}
                &lt;/p&gt;
              &lt;/section&gt;
&nbsp;
            &lt;/td&gt;
            &lt;td&gt;
              &lt;p className={`${styles.address}`}&gt;
                {item.sku}
              &lt;/p&gt;
            &lt;/td&gt;
            &lt;td&gt;
              &lt;p className={`${styles.address}`}&gt;
                {item.quantity}
              &lt;/p&gt;
            &lt;/td&gt;
            &lt;td className={styles.total}&gt;
              &lt;section&gt;
                &lt;p className={styles.subTitle}&gt;Subtotal:&lt;/p&gt;
                &lt;p
                  className={`${styles.info} ${styles.right}`}
                &gt;
                  {item.price}
                &lt;/p&gt;
&nbsp;
                &lt;p className={styles.subTitle}&gt;Frete:&lt;/p&gt;
                &lt;p
                  className={`${styles.info} ${styles.right}`}
                &gt;
                  {item.price}
                &lt;/p&gt;
&nbsp;
                &lt;p className={styles.subTitle}&gt;Valor total:&lt;/p&gt;
                &lt;p
                  className={`${styles.info} ${styles.right}`}
                &gt;
                  {item.price}
                &lt;/p&gt;
              &lt;/section&gt;
            &lt;/td&gt;
          &lt;/tr&gt;
        ))}
      &lt;/tbody&gt;
&nbsp;
      &lt;tfoot&gt;
        &lt;tr className={styles.tr}&gt;
          &lt;td /&gt;
          &lt;td /&gt;
          &lt;td /&gt;
          &lt;td className={styles.total}&gt;
            &lt;p
              className={`${styles.info} ${styles.left}`}
            &gt;
              {geral.quantity} unidades de {geral.items} items
            &lt;/p&gt;
&nbsp;
            &lt;section&gt;
              &lt;p className={styles.info}&gt;Subtotal:&lt;/p&gt;
              &lt;p
                className={`${styles.info} ${styles.right}`}
              &gt;
                {geral.totalPrice}
              &lt;/p&gt;
&nbsp;
              &lt;p className={styles.info}&gt;Frete total:&lt;/p&gt;
              &lt;p
                className={`${styles.info} ${styles.right}`}
              &gt;
                {geral.totalFreight}
              &lt;/p&gt;
&nbsp;
&nbsp;
              &lt;p className={styles.info}&gt;Valor total:&lt;/p&gt;
              &lt;p
                className={`${styles.info} ${styles.right}`}
              &gt;
                {geral.total}
              &lt;/p&gt;
            &lt;/section&gt;
&nbsp;
          &lt;/td&gt;
        &lt;/tr&gt;
      &lt;/tfoot&gt;
&nbsp;
    &lt;/table&gt;
  &lt;/Fragment&gt;
);
&nbsp;
ShipTableItems.propTypes = {
  items: array,
  geral: object,
};
&nbsp;
ShipTableItems.defaultProps = {
  items: [],
  geral: {},
};
&nbsp;
export default cssModules(ShipTableItems, styles, { allowMultiple: true });
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 11 2019 23:02:12 GMT+0000 (WET)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
