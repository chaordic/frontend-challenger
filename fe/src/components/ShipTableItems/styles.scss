@import '../../global-styles/settings.global.scss';

.table {
    display: grid;
    grid-column-gap: 1rem;
    grid-template-columns: 1fr;
}

.tr td:last-child {
    border-bottom: 1px solid #ccc;
}

.tr {
    display: grid;
    grid-column-gap: 1rem;
    grid-template-columns: 3fr 1fr 1fr 2fr;
    padding: 10px 20px 0 20px;
}

.mainProduct {
    display: grid;
    grid-column-gap: 1rem;
    grid-template-columns: 1fr 4fr;
}

.mainProduct img {
    width: 100%;
    height: 90%;
    border: 1px solid #ccc;
}

.background {
    background-color: #f8f8f8;
}