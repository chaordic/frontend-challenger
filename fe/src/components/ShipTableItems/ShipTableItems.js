import React, { Fragment } from 'react';
import { array, object } from 'prop-types';
import cssModules from 'react-css-modules';
import styles from './styles.scss';

/**
 * Format to portuguese shop status
 *
 * @param {Array} items Products
 * @param {Object} geral Object with consolidate data by items
 * @return {Component} Table of products
 */
const ShipTableItems = ({ items, geral }) => (
  <Fragment>
    <h1>Detalhes da entrega</h1>
    <table className={styles.table}>
      <thead className={`${styles.background}`}>
        <tr className={styles.tr}>
          <td>PRODUTO</td>
          <td>SKU</td>
          <td>QTD.</td>
          <td>PREÃ‡O</td>
        </tr>
      </thead>
      <tbody>
        {items.map((item, i) => (
          <tr key={i} className={`${styles.tr} ${(i + 1) % 2 === 0 ? `${styles.background}` : ''}`}>
            <td className={styles.mainProduct}>
              <img
                alt={item.name}
                title={item.name}
                src={item.image}
              />
              <section>
                <p className={`${styles.address}`}>
                  {item.name}
                </p>

                <p className={`${styles.address}`}>
                  {item.desc}
                </p>
              </section>

            </td>
            <td>
              <p className={`${styles.address}`}>
                {item.sku}
              </p>
            </td>
            <td>
              <p className={`${styles.address}`}>
                {item.quantity}
              </p>
            </td>
            <td className={styles.total}>
              <section>
                <p className={styles.subTitle}>Subtotal:</p>
                <p
                  className={`${styles.info} ${styles.right}`}
                >
                  {item.price}
                </p>

                <p className={styles.subTitle}>Frete:</p>
                <p
                  className={`${styles.info} ${styles.right}`}
                >
                  {item.price}
                </p>

                <p className={styles.subTitle}>Valor total:</p>
                <p
                  className={`${styles.info} ${styles.right}`}
                >
                  {item.price}
                </p>
              </section>
            </td>
          </tr>
        ))}
      </tbody>

      <tfoot>
        <tr className={styles.tr}>
          <td />
          <td />
          <td />
          <td className={styles.total}>
            <p
              className={`${styles.info} ${styles.left}`}
            >
              {geral.quantity} unidades de {geral.items} items
            </p>

            <section>
              <p className={styles.info}>Subtotal:</p>
              <p
                className={`${styles.info} ${styles.right}`}
              >
                {geral.totalPrice}
              </p>

              <p className={styles.info}>Frete total:</p>
              <p
                className={`${styles.info} ${styles.right}`}
              >
                {geral.totalFreight}
              </p>


              <p className={styles.info}>Valor total:</p>
              <p
                className={`${styles.info} ${styles.right}`}
              >
                {geral.total}
              </p>
            </section>

          </td>
        </tr>
      </tfoot>

    </table>
  </Fragment>
);

ShipTableItems.propTypes = {
  items: array,
  geral: object,
};

ShipTableItems.defaultProps = {
  items: [],
  geral: {},
};

export default cssModules(ShipTableItems, styles, { allowMultiple: true });
