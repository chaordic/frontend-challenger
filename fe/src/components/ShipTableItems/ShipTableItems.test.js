import React from 'react';
import renderer from 'react-test-renderer';
import ShipTableItems from './ShipTableItems';

const props = {
  items: [{
    color: 'Branco, Cinza',
    desc: 'Branco, Cinza, 39',
    image: 'http://image.url',
    name: 'Tenis Coca Coca Loretto - Feminino',
    price: 'R$100,00',
    quantity: 2,
    size: '39',
    sku: 'AR384675',
    stockType: 'PHYSICAL',
  }],
  geral: {
    quantity: 3, items: 2, totalFreight: 'R$0,00', totalPrice: 'R$200,00', total: 'R$200,00',
  },
};

describe('(Component) ShipTableItems', () => {
  it('should render an ShipTableItems with result', () => {
    const tree = renderer.create(<ShipTableItems {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
