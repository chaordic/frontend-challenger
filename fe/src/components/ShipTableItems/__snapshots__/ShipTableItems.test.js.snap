// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ShipTableItems should render an ShipTableItems with result 1`] = `
Array [
  <h1>
    Detalhes da entrega
  </h1>,
  <table>
    <thead
      className="undefined"
    >
      <tr>
        <td>
          PRODUTO
        </td>
        <td>
          SKU
        </td>
        <td>
          QTD.
        </td>
        <td>
          PREÃ‡O
        </td>
      </tr>
    </thead>
    <tbody>
      <tr
        className="undefined "
      >
        <td>
          <img
            alt="Tenis Coca Coca Loretto - Feminino"
            src="http://image.url"
            title="Tenis Coca Coca Loretto - Feminino"
          />
          <section>
            <p
              className="undefined"
            >
              Tenis Coca Coca Loretto - Feminino
            </p>
            <p
              className="undefined"
            >
              Branco, Cinza, 39
            </p>
          </section>
        </td>
        <td>
          <p
            className="undefined"
          >
            AR384675
          </p>
        </td>
        <td>
          <p
            className="undefined"
          >
            2
          </p>
        </td>
        <td>
          <section>
            <p>
              Subtotal:
            </p>
            <p
              className="undefined undefined"
            >
              R$100,00
            </p>
            <p>
              Frete:
            </p>
            <p
              className="undefined undefined"
            >
              R$100,00
            </p>
            <p>
              Valor total:
            </p>
            <p
              className="undefined undefined"
            >
              R$100,00
            </p>
          </section>
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td />
        <td />
        <td />
        <td>
          <p
            className="undefined undefined"
          >
            3
             unidades de 
            2
             items
          </p>
          <section>
            <p>
              Subtotal:
            </p>
            <p
              className="undefined undefined"
            >
              R$200,00
            </p>
            <p>
              Frete total:
            </p>
            <p
              className="undefined undefined"
            >
              R$0,00
            </p>
            <p>
              Valor total:
            </p>
            <p
              className="undefined undefined"
            >
              R$200,00
            </p>
          </section>
        </td>
      </tr>
    </tfoot>
  </table>,
]
`;
